tests/e2e/utils.js

```diff
@@ -456,3 +451,9 @@ export const describeOrSkip = ( condition ) =>
  * @param {boolean} condition Condition to execute test.
  */
 export const itOrSkip = ( condition ) => ( condition ? it : it.skip );
+
+export const insertAllProductBlock = async () => {
+	await insertBlock( 'Paragraph' );
+	await page.keyboard.type( '/All Products' );
+	await page.keyboard.press( 'Enter' );
+};
```

Comment:
Why does this require its own function? How come `insertBlock( 'All Products' )` does not work? Also keep in mind that `insertBlock` does other things such as waiting for the inserter to close and the focus to go on the content, which might be important to avoid race conditions and false negatives.

If a separate function needs to be used, please let's add some docs above explaining why and also let's make it more generic, like: `insertBlockViaKeyboardInput( blockname: string )`.

Reply:
I mentioned the reason for this in the description. After spending a good chunk of time debugging this, I couldn't find the cause of the issue, so this is just a workaround. 

<details>
<summary>The reason for a custom utility for inserting All Products block</summary>

For All Products block, the inserter doesn't close after inserting, which breaks the test. I can reproduce this issue on my local with or without Gutenberg, but on CI, the tests only fail with Gutenberg. I fixed this by creating a custom utility to insert the All Products block, which I'm not happy with it. But I couldn't find the root cause of the issue. I consider this is a hack/workaround and I'm more than happy if anyone has better solutions.

</details>

> If a separate function needs to be used, please let's add some docs above explaining why and also let's make it more generic, like: `insertBlockViaKeyboardInput( blockname: string )`.

Sounds good to me! I will try debugging the `All Products` block and `insertBlock` again. If we still need a separate function, I will refactor `insertAllProductBlock` following your suggestion.

Reply:
This is indeed super annoying. I'm very uncomfortable with pushing this workaround. I've said before that one of the problems with testing is that we treat it as second-class code. So we ship things that we'd never ship in production code, and we feed into the flakiness and unreliability by doing so. Sorts of like a [Broken Windows situation](https://en.wikipedia.org/wiki/Broken_windows_theory).

However, I understand the extreme frustration when something doesn't work with tests in a specific situation. I'm happy to pair a bit with you on this, and definitely to go the more generic function route if we don't find a solution.